src/test_utils.jl

```diff
@@ -753,6 +765,40 @@ function run_rrule!!_test_cases(rng_ctor, v::Val)
     return run_derived_rrule!!_test_cases(rng_ctor, v)
 end
 
+"""
```

Comment:
Should these functions be moved to DD?

Reply:
I haven't needed this utility in other libraries yet. But maybe eventually this could be put there as an option to `allow_unstable` or something?

For the moment, these are specific to Mooncake, to weaken the type stability checks, because a lot of the tests operate on unstable types to begin with and the type instabilities "make sense". But normally you would want to still throw an error for such circumstances.

Reply:
Maybe we could put it directly in the dispatch doctor test group though? It's tricky as we do need to inject this behaviour.

Reply:
Sounds good to keep them here if we move all this code into a new integration test for DD, as suggested below.